XenAdmin/Controls/SrPicker.cs

```diff
@@ -48,19 +48,22 @@ public enum SRPickerType { VM, InstallFromTemplate, Move, Copy, Migrate, LunPerV
 
         #region Private fields
 
+        private const int MAX_SCANS_PER_CONNECTION = 3;
+        private readonly ChangeableList<SrRefreshAction> _refreshQueue = new ChangeableList<SrRefreshAction>();
         private SRPickerType _usage = SRPickerType.VM;
         private VDI[] _existingVDIs;
         private IXenConnection _connection;
         private Host _affinity;
         private SR defaultSR;
         private readonly CollectionChangeEventHandler SR_CollectionChangedWithInvoke;
-        private volatile int _scanCount;
+        private readonly object _lock = new object();
 
         #endregion
 
         public SrPicker()
         {
             SR_CollectionChangedWithInvoke = Program.ProgramInvokeHandler(SR_CollectionChanged);
+            _refreshQueue.CollectionChanged += _refreshQueue_CollectionChanged;

```

Comment:
nit: event should be `RefreshQueue_CollectionChanged` to not break naming conventions. 

![image](https://user-images.githubusercontent.com/32554698/203101318-51ee0f7c-dc30-461e-9c52-682f1864d024.png)